nakadi/event_stream.py

```diff
@@ -7,72 +7,94 @@
 import sys
 
 
-def create_stream_generator(kafka_pool, topic, partition, start_from, opts):
+def create_stream_generator(kafka_pool, topic, cursors, opts):
+
+    topics = {}
+    for cursor in cursors:
+        topic_partition = (topic, int(cursor['partition']))
+        topics[topic_partition] = int(cursor['offset'])
+
+    partitions = [int(cursor['partition']) for cursor in cursors]
 
     start = datetime.datetime.now()
 
     # stream
     def generator():
-        messages_read = 0
+
         keep_alive_in_a_row = 0
-        current_batch = []
-        batch_start_time = datetime.datetime.now()
 
+        messages_read = 0
+        current_batch = dict(zip(partitions, [[] for _ in partitions]))
```

Comment:
my brain explodes! :D do use comments for initialisations like this :speak_no_evil: